kindcat_custom.cue

```diff
@@ -10,6 +92,68 @@ package kindsys
 Custom: S={
 	_sharedKind
 
-	lineage: { name: S.machineName }
+	// pluginID is the unique identifier of the plugin which owns this Custom kind
+	pluginID: =~"^([a-z][a-z0-9-]*[a-z0-9])$"
+
+	// isCRD is true if the `crd` trait is present in the kind.
+	isCRD: S.crd != _|_
+
+	lineage: { 
+		name: S.machineName
+	}
 	lineageIsGroup: false
+
+	if isCRD {
+		// If the crd trait is defined, the schemas in the lineage must follow the format:
+		// {
+		//     "metadata": CommonMetadata & {...string}
+		//     "spec": {...}
+		//     "status": {...}
+		// }
+		lineage: joinSchema: _crdSchema
+	}
+
+	// crd contains properties specific to converting this kind to a Kubernetes CRD.
+	// Unlike in Core, crd is optional and is used as a signaling mechanism for whether the kind is intended to be registered as a Kubernetes CRD 
+	// and/or a resource in a compatible API server. When present, additional structure is enforced on the kind's lineage's schemas.
+	// When absent, a lineage's schema has no restrictions as it is assumed that a CRD or similar resource type will not be generated from it.
+	// 
+	// TODO: rather than `crd`, should this trait be something more generic, as it really indicates more if a resource should be available in a
+	// kubernetes-compatible APIServer, not specifically as CRD (though that _is_ an implementation)
+	crd?: {
+		// groupOverride is an override that is used in the crd trait if present.
+		// If left empty, plugin.id is used to generate the group name
+		groupOverride?: =~"^([a-z][a-z0-9-]{0,32}[a-z0-9])$"
+
+		// _computedGroups is a list of groups computed from information in the plugin trait.
+		// The first element is always the "most correct" one to use.
+		// This field could be inlined into `group`, but is separate for clarity.
+		_computedGroups: [
+			if S.crd.groupOverride != _|_ {
+				strings.ToLower(S.crd.groupOverride) + ".apps.grafana.com",
```

Comment:
Is "apps" too specific? 

Will there be cases where Kinds are defined in the scope of other "extensions". Datasources, Apps, Plugins generally? d

Perhaps "ext.grafana.com"?

Reply:
I'm in favor of `ext.grafana.com`. Doesn't take up too much of the allotted 63 characters for the full GroupKind, and is clear enough about what it is (an extension vs something in core).

Reply:
`ext.grafana.com` works for me.

Reply:
Now uses `<group>.ext.grafana.com` (where group is typically the plugin ID). `groupOverride` can be set, which overrides the entire `crd.group` property, so there is a way to make the group something else if the need arises.